test/util/listener/listener.go

```diff
@@ -36,10 +38,18 @@ func (l *Listener) Accept() (net.Conn, error) {
 }
 
 func (l *Listener) Close() error {
-	if !l.closed {
+	l.RLock()
+	isListenerClosed := l.closed
+	l.RUnlock()
+
+	if !isListenerClosed {
 		close(l.c)
+
+		l.Lock()
 		l.closed = true
+		l.Unlock()
 	}
+
```

Comment:
There can still be a race condition here and you can have a double close with that code. You probably want to have a simple mutex which protects the whole function instead

Reply:
You're right. Wanted to optimize it with a RWMutex after the "basic" use of a Lock, but that is wrong here. Thanks!